backend/lcfs/web/api/organization/views.py

```diff
@@ -62,17 +72,57 @@ async def create_organization(
             logger.error("Internal Server Error: %s", e)
             raise HTTPException(
                 status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
-                detail="Internal Server Error"
+                detail="Internal Server Error",
             ) from e
 
+
+@router.get(
+    "/{organization_id}",
+    response_model=GetOrganizationResponse,
+    status_code=status.HTTP_200_OK,
+)
+async def get_organization(
+    organization_id: int,
+    db: AsyncSession = Depends(get_async_db),
+):
+    try:
+        org = await db.scalar(
+            select(Organization)
+            .options(
+                selectinload(Organization.org_status),
+                selectinload(Organization.org_address),
+                selectinload(Organization.org_attorney_address),
+            )
+            .where(Organization.organization_id == organization_id)
+        )
+
+        if org is None:
+            raise HTTPException(status_code=404, detail="org not found")
+
+        return org
+
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Internal Server Error: {str(e)}")
+
+
 @router.put("/{organization_id}", response_model=OrganizationSchema)
-async def update_organization(organization_id: int, organization_data: OrganizationUpdateSchema, db: AsyncSession = Depends(get_async_db)):
+async def update_organization(
```

Comment:
We don't have an edit view for org yet, so just a note that we'll have to test this again once the view is created.

Reply:
I think prashanth is working on the edit org side of things in a separate ticket.